### PR TITLE
Fix filter payload issue on conversation pagniation

### LINKED ISSUE #3739: API call is failing on selecting load more conversations after applying filter
**Describe the bug**

The API call is failing when we click to load more conversations after applying the filter.

**To Reproduce**

Steps to reproduce the behaviour:
1. Go to Dashboard and open the filter modal
2. Click on status -> open and apply filter and now you can see the filtered conversation.
3. Open dev tool and go to Network tab and verify the filters API and go back to dashboard
4. Scroll down to and select Load more conversations after that you can see the API call is getting an error.
5. You can see the steps in the below screen recording.

https://user-images.githubusercontent.com/64252451/149061127-5f48c359-9816-49db-b854-4a46e4dc6325.mov

**Desktop (please complete the following information):**
 - OS: Mac OS
 - Browser chrome
 - Version 98

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description

Fixes the conversations not loading bug when filters applied and load more button is clicked and lower cases all values if `filter_operator` is `equal_to` or `not_equal_to`

Fixes #3739 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: app/javascript/dashboard/helper/filterQueryGenerator.js [hunk 1]
```diff
@@ -1,5 +1,19 @@
+function lowerCaseValues(operator, values) {
```
Comment (nithindavid) [inline app/javascript/dashboard/helper/filterQueryGenerator.js:1]:
```suggestion
const lowerCaseValues = (operator, values) => {
```

File: app/javascript/dashboard/helper/filterQueryGenerator.js [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+      return val;
+    });
+  }
+  return values;
+}
+
 const generatePayload = data => {
-  let payload = data.map(item => {
+  // Make a copy of data to avoid vue data reactivity issues
+  const filters = JSON.parse(JSON.stringify(data));
```
Comment (nithindavid) [inline app/javascript/dashboard/helper/filterQueryGenerator.js:15]:
```suggestion
  const filters =  [...data];
```

Comment (nithindavid) [reply app/javascript/dashboard/helper/filterQueryGenerator.js:15]: Does this work?

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Review by pranavrajs — APPROVED** [review]

File: app/javascript/dashboard/helper/filterQueryGenerator.js [hunk 3]
```diff
@@ -8,11 +8,11 @@ function lowerCaseValues(operator, values) {
     });
   }
   return values;
-}
+};
 
 const generatePayload = data => {
   // Make a copy of data to avoid vue data reactivity issues
-  const filters = JSON.parse(JSON.stringify(data));
+  const filters = [...data];
```
Comment (pranavrajs) [inline app/javascript/dashboard/helper/filterQueryGenerator.js:15]: This won't work. It doesn't make a copy of the object. It just changes reference to the array

Comment (fayazara) [reply app/javascript/dashboard/helper/filterQueryGenerator.js]: Reverting, found out spread only does a shallow copy with root level keys

**Pull request merged**